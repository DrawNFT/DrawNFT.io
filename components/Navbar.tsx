import { FC } from 'react';
import Link from 'next/link';
import { useWeb3Handler } from './utils/useWeb3Handler';

const NavBar: FC = () => {
  const { account, nftContract } = useWeb3Handler();

  return (
    <nav className="w-full border-b border-gray-200 px-2 sm:px-4 py-2.5 dark:bg-violet-900 sticky top-0 z-40">
      <div className="flex flex-wrap items-center justify-between">
        <span className="self-center text-xl font-semibold whitespace-nowrap dark:text-white flex gap-4">
          <Link href="/">Home</Link>
          <Link href="/whitepaper">White Paper</Link>
          <Link
            target="_blank"
            rel="noopener noreferrer"
            href="https://github.com/DrawNFT/DrawNFT.io"
          >
            Github
          </Link>
          <Link
            target="_blank"
            rel="noopener noreferrer"
            href="https://twitter.com/DrawNFTio"
          >
            Twitter
          </Link>
        </span>

        <span className="self-center text-xl font-semibold whitespace-nowrap dark:text-white flex gap-10">
          {account &&
            account?.toLowerCase() ===
              process.env.WITHDRAW_ACCOUNT?.toLowerCase() && (
              <button
                onClick={async () => {
                  await nftContract?.withdrawMintPayments();
                }}
              >
                Withdraw Payments
              </button>
            )}

          {account ? (
            <p>{account}</p>
          ) : (
            <button
              onClick={() => {
                alert('You need a web3 wallet to intract with this app!');
              }}
            >
              Connect Wallet
            </button>
          )}
        </span>
      </div>
    </nav>
  );
};

export default NavBar;
