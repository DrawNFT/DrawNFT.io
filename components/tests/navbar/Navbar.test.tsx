import Navbar from '../../../components/Navbar';
import { render } from '@testing-library/react';

const useWeb3HandlerFn = jest.fn();

jest.mock('../../../components/utils/useWeb3Handler', () => ({
  __esModule: true,
  ...jest.requireActual('../../../components/utils/useWeb3Handler'),
  useWeb3Handler: () => useWeb3HandlerFn(),
}));

describe('Navbar Component Test', () => {
  it('Renders correctly without env', () => {
    // given
    useWeb3HandlerFn.mockReturnValue({
      account: 'account',
      nftContract: 'nftContract',
    });

    // when
    const { container, getByText } = render(<Navbar />);

    // then
    expect(container).toMatchSnapshot();
    expect(getByText('account').closest('a')?.getAttribute('href')).toBe(
      'https://opensea.io/account'
    );
  });

  it('Renders correctly with button click without account', () => {
    // given
    useWeb3HandlerFn.mockReturnValue({
      account: undefined,
      nftContract: 'nftContract',
    });
    jest.spyOn(window, 'alert').mockImplementation(() => {});

    // when
    const { container } = render(<Navbar />);

    const button = container.querySelector('button');
    button?.click();

    // then
    expect(container).toMatchSnapshot();
    expect(window.alert).toBeCalledWith(
      'You need a web3 wallet to intract with this app!'
    );
  });

  it('Renders correctly with withdraw payments button', () => {
    // given
    const nftContract = jest.fn();
    useWeb3HandlerFn.mockReturnValue({
      account: 'account',
      nftContract: { withdrawMintPayments: () => nftContract() },
    });
    process.env.WITHDRAW_ACCOUNT = 'account';

    // when
    const { container, getByText } = render(<Navbar />);

    const button = container.querySelector('button');
    button?.click();

    // then
    expect(container).toMatchSnapshot();
    expect(nftContract).toBeCalledTimes(1);
    expect(getByText('Withdraw Payments')).not.toBe(undefined);
  });

  it('render with account does not match with the process', () => {
    // given
    const nftContract = jest.fn();
    useWeb3HandlerFn.mockReturnValue({
      account: 'account',
      nftContract: { withdrawMintPayments: () => nftContract() },
    });
    process.env.WITHDRAW_ACCOUNT = 'not_account';

    // when
    const { container, getByText } = render(<Navbar />);

    // then
    expect(container).toMatchSnapshot();
    expect(() => getByText('Withdraw Payments')).toThrow();
  });
});
