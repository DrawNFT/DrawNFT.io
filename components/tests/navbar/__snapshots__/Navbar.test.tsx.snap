// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Test Renders correctly with button click without account 1`] = `
<div>
  <nav
    class="w-full border-b border-gray-200 px-2 sm:px-4 py-2.5 bg-violet-900 sticky top-0 z-40"
  >
    <div
      class="flex flex-wrap items-center justify-between"
    >
      <span
        class="self-center text-xl font-semibold whitespace-nowrap text-white flex gap-4"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/whitepaper"
        >
          White Paper
        </a>
        <a
          href="https://github.com/DrawNFT/DrawNFT.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          Github
        </a>
        <a
          href="https://twitter.com/DrawNFTio"
          rel="noopener noreferrer"
          target="_blank"
        >
          Twitter
        </a>
      </span>
      <span
        class="self-center text-xl font-semibold whitespace-nowrap text-white flex gap-10"
      >
        <button>
          Connect Wallet
        </button>
      </span>
    </div>
  </nav>
</div>
`;

exports[`Navbar Component Test Renders correctly with withdraw payments button 1`] = `
<div>
  <nav
    class="w-full border-b border-gray-200 px-2 sm:px-4 py-2.5 bg-violet-900 sticky top-0 z-40"
  >
    <div
      class="flex flex-wrap items-center justify-between"
    >
      <span
        class="self-center text-xl font-semibold whitespace-nowrap text-white flex gap-4"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/whitepaper"
        >
          White Paper
        </a>
        <a
          href="https://github.com/DrawNFT/DrawNFT.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          Github
        </a>
        <a
          href="https://twitter.com/DrawNFTio"
          rel="noopener noreferrer"
          target="_blank"
        >
          Twitter
        </a>
      </span>
      <span
        class="self-center text-xl font-semibold whitespace-nowrap text-white flex gap-10"
      >
        <button>
          Withdraw Payments
        </button>
        <a
          href="https://opensea.io/account"
          rel="noopener noreferrer"
          target="_blank"
        >
          account
        </a>
      </span>
    </div>
  </nav>
</div>
`;

exports[`Navbar Component Test Renders correctly without env 1`] = `
<div>
  <nav
    class="w-full border-b border-gray-200 px-2 sm:px-4 py-2.5 bg-violet-900 sticky top-0 z-40"
  >
    <div
      class="flex flex-wrap items-center justify-between"
    >
      <span
        class="self-center text-xl font-semibold whitespace-nowrap text-white flex gap-4"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/whitepaper"
        >
          White Paper
        </a>
        <a
          href="https://github.com/DrawNFT/DrawNFT.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          Github
        </a>
        <a
          href="https://twitter.com/DrawNFTio"
          rel="noopener noreferrer"
          target="_blank"
        >
          Twitter
        </a>
      </span>
      <span
        class="self-center text-xl font-semibold whitespace-nowrap text-white flex gap-10"
      >
        <a
          href="https://opensea.io/account"
          rel="noopener noreferrer"
          target="_blank"
        >
          account
        </a>
      </span>
    </div>
  </nav>
</div>
`;

exports[`Navbar Component Test render with account does not match with the process 1`] = `
<div>
  <nav
    class="w-full border-b border-gray-200 px-2 sm:px-4 py-2.5 bg-violet-900 sticky top-0 z-40"
  >
    <div
      class="flex flex-wrap items-center justify-between"
    >
      <span
        class="self-center text-xl font-semibold whitespace-nowrap text-white flex gap-4"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/whitepaper"
        >
          White Paper
        </a>
        <a
          href="https://github.com/DrawNFT/DrawNFT.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          Github
        </a>
        <a
          href="https://twitter.com/DrawNFTio"
          rel="noopener noreferrer"
          target="_blank"
        >
          Twitter
        </a>
      </span>
      <span
        class="self-center text-xl font-semibold whitespace-nowrap text-white flex gap-10"
      >
        <a
          href="https://opensea.io/account"
          rel="noopener noreferrer"
          target="_blank"
        >
          account
        </a>
      </span>
    </div>
  </nav>
</div>
`;
